js/viewer/annotation/base.js

```diff
@@ -95,7 +101,7 @@ function Base(attrs){
             rst += (key + '="' + attributeList[key] + '" ');
         });
 
-        rst += "></" + tag + ">";
+        rst += "></" + tag + ">" + markerDef;
```

Comment:
Didn't we say we want the definitions to be attached to the group and not each individual drawing? If I add multiple arrows of the same styles, you're adding multiple definitions that are doing the same thing.

Reply:
Moved the logic to add the marker definitions from base.js to annotation-group.js. Updated both the files and added appropriate comments.